### PR TITLE
[feat] support Cloudwatch Loggroup encryption at REST using KmsKey

### LINKED ISSUE #31788: CloudWatch logGroup support encryption at rest using KmsKey
### Component(s)

internal/aws

### Is your feature request related to a problem? Please describe.

Organizations maintain rigorous standards regarding data encryption. In my organization, it's imperative to encrypt all eligible data on AWS at rest utilizing our designated KMS key.

However, there's a significant limitation within the [internal/aws/cwlogs](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/aws/cwlogs) package designed for AWS CloudWatch interaction. This limitation arises from its inability to create CloudWatch LogGroups while specifying a custom KMS key, aside from the default option. Consequently, tools like [exporter/awsemfexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsemfexporter), which rely on this package, are unable to encrypt log data within CloudWatch LogGroups at rest using a custom KMS key.

### Describe the solution you'd like

Enhance the functionality of the internal/aws/cwlogs package to enable encryption of CloudWatch LogGroup data at rest using a specified KMS key. This addition will empower tools such as exporter/awsemfexporter to encrypt log data within CloudWatch LogGroups using a custom KMS key.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by edwin-Marrima**
**Link to tracking Issue:** <Issue number if applicable>
closes #31788

### CODE REVIEW

**Review by Frapschen — COMMENTED** [review]
please add a changelog

**Comment by edwin-Marrima** [comment]
@Frapschen any update on this?

File: CHANGELOG-API.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ If you are looking for user-facing changes, check out [CHANGELOG.md](./CHANGELOG
 
 <!-- next version -->
 
+## v0.97.0
```
Comment (Frapschen) [inline CHANGELOG-API.md:10]: please refer to this https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-a-changelog-entry to create changelog entry

**Review by Frapschen — APPROVED** [review]

**Comment by Frapschen** [comment]
@edwin-Marrima  Should awsemfexporter need to change something to adopt this change? if so please create another issue to track it

**Comment by edwin-Marrima** [comment]
The idea is to update `awsemfexporter`.  Will create new issue as suggested.

**Review by Frapschen — APPROVED** [review]
LGTM

**Comment by edwin-Marrima** [comment]
@Aneurysm9 any feedback on this?

**Comment by trc-ikeskin** [comment]
Any news on this?

**Comment by trc-ikeskin** [comment]
@Aneurysm9 @mxiamxia any chance you could review this? The possibility to auto-create encrypted CW log groups is badly needed to use the exporter based on this in production!

**Pull request closed without merging**